src/olympia/addons/indexers.py

```diff
@@ -5,20 +5,165 @@
 
 import olympia.core.logger
 from olympia import amo
-from olympia.amo.indexers import BaseSearchIndexer
 from olympia.amo.utils import attach_trans_dict
 from olympia.amo.celery import create_chunked_tasks_signatures
+from olympia.amo.utils import to_language
+from olympia.constants.search import SEARCH_LANGUAGE_TO_ANALYZER
 from olympia.lib.es.utils import create_index
 from olympia.versions.compare import version_int
 
 
 log = olympia.core.logger.getLogger('z.es')
 
 
-class AddonIndexer(BaseSearchIndexer):
-    """Fields we don't need to expose in the results, only used for filtering
-    or sorting."""
+class AddonIndexer:
```

Comment:
Because we only have one "document type" (*) left I moved the contents of `BaseSearchIndexer` into `AddonIndexer`.

(*) Technically not called a document type anymore since this concept is going away - that's part of the changes needed for 7.x compatibility. If in the future we want to re-add multiple types, they'll need to live on different indexes.